express-api/src/services/notifications/notificationServices.ts

```diff
@@ -0,0 +1,206 @@
+import { AppDataSource } from '@/appDataSource';
+import { Agency } from '@/typeorm/Entities/Agency';
+import { NotificationQueue } from '@/typeorm/Entities/NotificationQueue';
+import { NotificationTemplate } from '@/typeorm/Entities/NotificationTemplate';
+import { Project } from '@/typeorm/Entities/Project';
+import { ProjectStatusNotification } from '@/typeorm/Entities/ProjectStatusNotification';
+import { User } from '@/typeorm/Entities/User';
+import { UUID, randomUUID } from 'crypto';
+import nunjucks from 'nunjucks';
+import { IsNull } from 'typeorm';
+import chesServices, {
+  EmailBody,
+  EmailEncoding,
+  EmailPriority,
+  IEmail,
+} from '../ches/chesServices';
+import { SSOUser } from '@bcgov/citz-imb-sso-express';
+
+interface AccessRequestData {
+  FirstName: string;
+  LastName: string;
+}
+
+export enum NotificationStatus {
+  Accepted = 0,
+  Pending = 1,
+  Cancelled = 2,
+  Failed = 3,
+  Completed = 4,
+}
+
+export enum NotificationAudience {
+  ProjectOwner = 'ProjectOwner',
+  OwningAgency = 'OwningAgency',
+  Agencies = 'Agencies',
+  ParentAgencies = 'ParentAgencies',
+  Default = 'Default',
+  WatchingAgencies = 'WatchingAgencies',
+}
+
+const Title = 'PIMS';
+const Uri = '';
+
+const flattenProjectProperties = (project: Project) => {
+  const flattenedProperties = project.ProjectProperties.map((projectProperty) => {
+    if (projectProperty.Building != null) {
+      return {
+        ...projectProperty.Parcel,
+        Type: 'Parcel',
+      };
+    } else {
+      return {
+        ...projectProperty.Building,
+        Type: 'Building',
+      };
+    }
+  });
+  return {
+    ...project,
+    Properties: flattenedProperties,
+  };
+};
+
+const generateAccessRequestNotification = async (
+  accessRequest: AccessRequestData,
+  templateId: number,
```

Comment:
Does this need a template Id specified? Isn't this always the notification template for access requests?

Reply:
I think yes because there are two separate access request templates in the db. This is similar to how the old API handles it too.